presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java

```diff
@@ -359,7 +366,7 @@ private synchronized void scheduleTaskIfNecessary(TaskHandle taskHandle)
         // immediately schedule a new split for this task.  This assures
         // that a task gets its fair amount of consideration (you have to
         // have splits to be considered for running on a thread).
-        if (taskHandle.getRunningLeafSplits() < minimumNumberOfDriversPerTask) {
+        if (taskHandle.getRunningLeafSplits() < min(minimumNumberOfDriversPerTask, taskHandle.getMaxDriversPerTask().orElse(Integer.MAX_VALUE))) {
```

Comment:
For now we just want a user specified cap on the per task driver count. 

I think the `minimumNumberOfDriverPerTask` is `guaranteedNumberOfDriverPerTask`. We allow user to set a cap which is even lower than the system guaranteed driver count per task so that user can be `nice` and tell the system to run certain queries slower.

Reply:
This piece of code reads in a weird way, but IIUC the reason for this change is to be able to reduce the guaranteed number of drivers per task through the new session property. Renaming `minimumNumberOfDriversPerTask` to `guaranteedNumberOfDriverPerTask` would make this less confusing, so I feel like we should do the rename as part of this PR (in a separate commit before this commit), what do you think?